components/03-organisms/galleries/media-grid/yds-media-grid.twig

```diff
@@ -0,0 +1,34 @@
+{#
+ # Available Variables:
+ # - media_grid__heading (optional)
+ # - media_grid__variation: basic (default), interactive
+ #
+ # Available Blocks:
+ # - media_grid__items
+ #}
+
+{% set media_grid__base_class = 'media-grid' %}
+
+{% set media_grid__attributes = {
+  'data-component-width': 'feature',
+  'data-media-grid-variation': media_grid__variation|default('basic'),
+} %}
+
+<div {{ bem(media_grid__base_class) }} {{ add_attributes(media_grid__attributes) }}>
+  <div {{ bem('inner', [], media_grid__base_class) }}>
+    {% if media_grid__heading %}
+      {% include "@atoms/typography/headings/yds-heading.twig" with {
+        heading__level: '2',
+        heading__blockname: media_grid__base_class,
+        heading: media_grid__heading,
+      } %}
+    {% endif %}
+    <ul {{ bem('items', [], media_grid__base_class) }}>
```

Comment:
This seems like it should be embedding the generic list component instead of using the `<ul>` directly. Is your thinking here that doing so just complicates this component?

Reply:
Our other collections are currently written this way. I don't see any value add at the moment for the added complexity that would come with implementing the list component.

If we did, this:
```
{% set media_grid__base_class = 'media-grid' %}

{% set media_grid__attributes = {
  'data-component-width': 'feature',
  'data-media-grid-variation': media_grid__variation|default('basic'),
  'class': bem(media_grid__base_class),
} %}

<div {{ add_attributes(media_grid__attributes) }}>
  <div {{ bem('inner', [], media_grid__base_class) }}>
    {% if media_grid__heading %}
      {% include "@atoms/typography/headings/yds-heading.twig" with {
        heading__level: '2',
        heading__blockname: media_grid__base_class,
        heading: media_grid__heading,
      } %}
    {% endif %}
    <ul {{ bem('items', [], media_grid__base_class) }}>
      {% block media_grid__items %}
        {% for item in media_grid__items %}
          {% include "@organisms/galleries/media-grid/_yds-media-grid-item.twig" %}
        {% endfor %}
      {% endblock %}
    </ul>
  </div>
</div>
```

would become:
```
{% set media_grid__base_class = 'media-grid' %}

{% set media_grid__attributes = {
  'data-component-width': 'feature',
  'data-media-grid-variation': media_grid__variation|default('basic'),
  'class': bem(media_grid__base_class),
} %}

{% set media_grid__items %}
  {% block media_grid__items %}
    {% for item in media_grid__items %}
      {% include "@organisms/galleries/media-grid/_yds-media-grid-item.twig" %}
    {% endfor %}
  {% endblock %}
{% endset %}

<div {{ add_attributes(media_grid__attributes) }}>
  <div {{ bem('inner', [], media_grid__base_class) }}>
    {% if media_grid__heading %}
      {% include "@atoms/typography/headings/yds-heading.twig" with {
        heading__level: '2',
        heading__blockname: media_grid__base_class,
        heading: media_grid__heading,
      } %}
    {% endif %}
    {% embed "@atoms/lists/yds-list.twig" with {
      list__base_class: 'items',
      list__blockname: media_grid__base_class,
    } %}
      {% block list__content %}
        {{ media_grid__items }}
      {% endblock %}
    {% endembed %}
  </div>
</div>
```

Adding the extra "intermediate block" is a sort of anti-pattern. It works, but it's not real clean, and can make things more confusing. i'm not going to do the refactor now. But if we ever come across a reason to have a shared "list" component across the three collections (card-collection, custom-card-collection, and media-grid) then we can refactor them together.

Reply:
That makes sense to me. Although, I do think we should strive to reuse things as much as possible I see what you're saying about it be a bit of an anti-pattern. I think we may need to fundamentally rethink the list component in general so it's easier to re-use and doesn't require this added external block. But this is future-Brian/Callin's problem. 

Approving :+1: